autoload/neomake/makers/ft/javascript.vim

```diff
@@ -64,11 +64,11 @@ function! neomake#makers#ft#javascript#rjsx() abort
         \ .'(progn(package-initialize)(require ''rjsx-mode)'
         \ .'  (setq js2-include-node-externs t js2-include-rhino-externs t js2-include-browser-externs t js2-strict-missing-semi-warning nil)'
         \ .'  (rjsx-mode)(js2-reparse)(js2-display-error-list)'
-        \ .'  (princ(with-current-buffer "*js-lint*"'
-        \ .'    (replace-regexp-in-string "^" (concat (elt command-line-args 1) " ")'
-        \ .'      (buffer-substring-no-properties(point-min)(point-max)))))(terpri))'],
+        \ .'  (princ(replace-regexp-in-string "^" (concat buffer-file-name " ")'
+        \ .'  (with-current-buffer "*js-lint*" (buffer-substring-no-properties(point-min)(point-max)))))(terpri))'],
         \ 'errorformat': '%f\ line\ %l:\ %m',
         \ 'append_file': 0,
+        \ 'remove_invalid_entries': 1,
```

Comment:
Yes, it will be removed by Vim then already, and should mean less filtering/processing inside of Neomake.

`%-G%.%#` means to match `.*` (`%.%#`) as a general message (`%G`), but discard it (`-`).